### ISSUE TITLE
Split the project into 2 repositories: libtransmission (core) and The GUI clients as Transmission 

### ISSUE THREAD

**Issue by Pentaphon** [transmission/transmission#7190]
### What is the issue?

Per Coeur's suggestion in https://github.com/transmission/transmission/issues/7058#issuecomment-2296655083

> The admins could also split the project in two repositories:
> 
>     The core as libtransmission
>     The GUI clients as Transmission
> 
> That way, they can grant "merge" access to only half the project. And calling it "libtransmission" could attract new projects to adopt it for their own clients.

I made this issue so we can discuss this and track the progress of making it happen. 

### Which application of Transmission?

None

### Which version of Transmission?

N/A

**Comment by killemov** [comment]
Is merge access rights the only argument for splitting it up in 2 repos? (With Subversion we had partial/subtree update rights, is that not possible for Git in some way?) For me a switch for building a separate(dynamic!)/integrated libtransmission would be fine. An integrated version has the benefits for code optimization.

**Comment by Pentaphon** [comment]
> Is merge access rights the only argument for splitting it up in 2 repos?

Not quite. The other major torrent projects do it this way, such as the 2 libtorrent projects. Splitting it up could make things easier for new devs to find libtransmission and use it while also possibly contributing to it. I think people basically want 2 things out of Transmission right now and that's:

1. Better performance (esp. async nonblocking disk IO)
2. More features in the GUI clients

So splitting them will at least accelerate number 1.

**Comment by ckerr** [comment]
It's pretty common right now to make changes to libtransmission/ that ripple into the other directories. Keeping everything in a monorepo lets us keep everything in sync. Recent examples: #8341, #8257, #8256, #8253, #8229. These are all from the past week.

This isn't a bad idea, but I don't think it's practical for now.
